### PR TITLE
Apply UI/UX suggestions to page tree and add tutorial pop-up

### PR DESCRIPTION
**Pull request by ulliholtgrave**
### Short description

This PR applies the changes suggested by Nicola and adds a generic approach to add tutorial pop-ups to our views.

### Proposed changes

- Restructure Tree Table
- Add generic tutorial template (as requested in #57 )
- Add tree view tutorial

### Resolved issues

Fixes: #844 
Fixes: #57

### CODE REVIEW

File: integreat_cms/cms/templates/pages/page_tree.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
 {% with filter_form.filters_visible as filters_visible %}
 <div class="table-header">
     <div class="flex flex-wrap justify-between gap-4">
-        <h1 class="heading">{% trans 'Page Tree' %}</h1>
+        <h1 class="heading">{% trans 'Page Tree' %} <button data-show-tutorial="page_tree_tutorial"><i data-feather="help-circle"></i></button></h1>
```
Comment (timoludwig) [inline integreat_cms/cms/templates/pages/page_tree.html:14]:
```suggestion
        <h1 class="heading">{% trans 'Page Tree' %} <button data-show-tutorial="page_tree_tutorial" class="hover:text-blue-500"><i data-feather="help-circle" class="align-baseline"></i></button></h1>
```
I think this looks a bit better

File: integreat_cms/cms/views/settings/dismiss_tutorial_view.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from django.utils.decorators import method_decorator
+from django.contrib.auth.decorators import login_required
+from django.http.response import HttpResponseRedirect
+from django.views.generic.base import RedirectView
+
+
+@method_decorator(login_required, name="dispatch")
+class DismissTutorial(RedirectView):
+    r"""
```
Comment (timoludwig) [inline integreat_cms/cms/views/settings/dismiss_tutorial_view.py:9]:
```suggestion
    """
```

File: integreat_cms/cms/views/settings/dismiss_tutorial_view.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.http.HttpResponseRedirect
+        """
+        redirect_url = request.POST.get("redirect_url")
+        if not redirect_url:
+            redirect_url = "/"
```
Comment (timoludwig) [inline integreat_cms/cms/views/settings/dismiss_tutorial_view.py:32]:
```suggestion
        redirect_url = request.POST.get("redirect_url", "/")
```
QueryDicts can retrieve keys with fallback values.

File: integreat_cms/cms/views/settings/dismiss_tutorial_view.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+        if request.POST.get("dismiss_tutorial") and request.POST.get("tutorial_id"):
+            tutorial_id = request.POST.get("tutorial_id")
+            if tutorial_id == "page_tree_tutorial":
+                request.user.page_tree_tutorial = True
+            request.user.save()
```
Comment (timoludwig) [inline integreat_cms/cms/views/settings/dismiss_tutorial_view.py:38]:
```suggestion
        tutorial_id = request.POST.get("tutorial_id")
        if request.POST.get("dismiss_tutorial") and tutorial_id:
            if tutorial_id == "page_tree_tutorial":
                request.user.page_tree_tutorial = True
            request.user.save()
```

**Review by timoludwig — COMMENTED** [review]
Thanks! :rocket:
See my comment below for my detailed feedback.

**Comment by timoludwig** [comment]
Oh, and one thing I forgot: On my device, the popup is actually larger than the screen but there is no scrolling, so I can not click onto the "Understood" button unless I zoom out. Maybe we need a bit of responsiveness here?

File: integreat_cms/cms/templates/tutorials/page_tree.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+<!-- Pop-Up Content -->
+<div class="px-4">
+    <p class="mt-4 text-s max-w-fit text-gray-600">{% trans 'In our new tree view, we have merged the two previous views into one which includes all previous functions. In this view you are provided with the following functionalities:' %}</p>
+    <ul class="list-disc my-2 pl-5 text-s text-gray-600">
+        <li>{% trans 'Viewing the page hierarchy in the left column.' %}</li>
+        <li>{% trans 'Changing the hierarchy by drag & drop.' %}</li>
+        <li>{% trans 'Viewing the different statuses of the translations of a page.' %}</li>
+        <li>{% trans "Open the page editor by clicking on a page." %}</li>
+    </ul>
+    <img src="{% static 'images/page_tree_tutorial.gif' %}" class="m-auto my-4" />
```
[the remark this answers is not in the document]
Comment (ulliholtgrave) [reply integreat_cms/cms/templates/tutorials/page_tree.html]: Has been added

File: integreat_cms/cms/templates/tutorials/page_tree.html [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{% extends "../generic_tutorial_dialog.html" %}
+{% load i18n %}
+{% load static %}
+
+{% block content %}
+<!-- Pop-Up Header -->
+<div class="flex items-center bg-blue-500 w-full rounded min-h-12">
+    <i data-feather="help-circle" class="text-white ml-2"></i>
+    <h2 class="flex items-center font-semibold leading-normal pl-2 text-white text-2xl"> {% trans 'Introduction - Page Tree' %}</h2>
```
Comment (timoludwig) [inline integreat_cms/cms/templates/tutorials/page_tree.html:9]:
```suggestion
    <h2 class="flex items-center font-semibold leading-normal pl-2 text-white text-xl"> {% trans 'Introduction - Page Tree' %}</h2>
```
I think `text-2xl` is a bit large

File: integreat_cms/cms/templates/tutorials/page_tree.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{% extends "../generic_tutorial_dialog.html" %}
+{% load i18n %}
+{% load static %}
+
+{% block content %}
+<!-- Pop-Up Header -->
+<div class="flex items-center bg-blue-500 w-full rounded min-h-12">
```
Comment (timoludwig) [inline integreat_cms/cms/templates/tutorials/page_tree.html:7]:
```suggestion
<div class="flex items-center bg-blue-500 w-full rounded h-12">
```

File: integreat_cms/cms/templates/generic_tutorial_dialog.html [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+            {% block content %}{% endblock %}
+            {% if tutorial_id %}
+                <form method="post" action="{% url 'dismiss_tutorial' %}" data-tutorial-overlay-form class="p-2">
+                    {% csrf_token %}
+                    <input type="hidden" name="tutorial_id" value="{{ tutorial_id }}" />
+                    <input type="hidden" name="redirect_url" value="{{ request.get_full_path }}" />
+                    
+                    <div class="flex px-2 gap-2 justify-between">
+                        <label for="dismiss-tutorial" class="flex font-normal text-s text-gray-600 items-center">
+                            <input id="dismiss-tutorial" type="checkbox" name="dismiss_tutorial" value="{{ request.user.page_tree_tutorial_seen }}" />
```
Comment (timoludwig) [inline integreat_cms/cms/templates/generic_tutorial_dialog.html:15]:
```suggestion
                            <input id="dismiss-tutorial" type="checkbox" name="dismiss_tutorial" {% if request.user.page_tree_tutorial_seen %}checked{% endif %} />
```
Ok, my previous suggestion didn't work as I expected^^

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 9]
```diff
@@ -5159,6 +5177,36 @@ msgstr "Bild/PNG"
 msgid "Table Document/CSV"
 msgstr "Tabellendokument/CSV"
 
+#: cms/templates/tutorials/page_tree.html:9
+msgid "Introduction - Page Tree"
+msgstr "Einführung in den neuen Seiten-Baum"
+
+#: cms/templates/tutorials/page_tree.html:13
+msgid ""
+"In our new tree view, we have merged the two previous views into one which "
```
Comment (svenseeberg) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:5186]: `We merged "All pages" and "Tree View" into one single view. In this view, you can do the following:`

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 10]
```diff
@@ -5159,6 +5177,36 @@ msgstr "Bild/PNG"
...
+"includes all previous functions. In this view you are provided with the "
+"following functionalities:"
+msgstr ""
+"In unser neuen Baumansicht haben wir zwei bisherigen Seiten-Ansichten in "
```
Comment (svenseeberg) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:5190]: `Wir haben "Alle Seiten" und "Baumansicht" in eine Ansicht zusammengeführt. In dieser Ansicht können folgende Aktionen durchgeführt werden:`

**Review by svenseeberg — APPROVED** [review]

**Pull request merged**